### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by kazalt**

### CODE REVIEW

File: attribute.js [hunk 1]
```diff
@@ -46,14 +46,22 @@ function constructorValues(value) {
...
   });
 }
 
 function constructorTriggers() {
-  const onChangeCallbacks = [];
-  this.onChange = (callback) => onChangeCallbacks.push(callback);
-  this.trigger = () => onChangeCallbacks.forEach((callback) => callback.call(this, this.value, this.getOriginalValue()));
+  const eventCallbacks = {};
+
+  this.on = (eventType, callback) => {
```
Comment (oudirichi) [inline attribute.js]: The event `on` and `trigger` should be on a separated class named something like `Event` and `Attribute` `extends` it

File: attribute-mixins/validation.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          }
+        });
+      }
+    }
+
+    return errors;
+  }
+};
+
+export default ValidationMixin;
```
Comment (oudirichi) [inline attribute-mixins/validation.js]: it's weird that we have mixin validation that export a ValidationMixin and in attribute-mixins/validation.js we return a ValidationMixin. I think we should rename this class to AttributeValidationMixin

File: test/attribute-mixins/validation.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+        it('should add required errors', () => {
+          attribute.validate();
+          expect(attribute.errors).to.deep.equal({ required: false });
+        });
+      });
+
+      context('with custom validations', () => {
+        beforeEach(() => {
+          Validator.failingValidation = () => false;
```
Comment (oudirichi) [inline test/attribute-mixins/validation.js]: should be `passingValidation` because it's always valid

File: attribute-mixins/validation.js (hunk 2 above)
Comment (kazalt) [reply attribute-mixins/validation.js]: When you import it you can name it like you want. Still i will rename it

**Pull request merged**